### PR TITLE
EZS-1393: Fixed select location js error making it impossible to change the location

### PR DESCRIPTION
**Pull request by sunpietro**
https://jira.ez.no/browse/EZS-1393
 
The issue was reported by the community.

### CODE REVIEW

**Review by dew326 — APPROVED** [review]

**Review by micszo — APPROVED** [review]
Retest OK. 👍

**Comment by andrerom** [comment]
Will this break 1.7? /cc @dpobel 

Asking as this repo is not branched, so we would then first need to branch of version for 1.7 before we can apply this.

**Comment by sunpietro** [comment]
@andrerom I agree with you. It will be the safest option.

**Comment by andrerom** [comment]
@sunpietro There are two main options here:
A. You somehow feature detect if this is 1.7 or 1.8, allowing us to support both with same code base _(if not to hard to do, this is my personal preference)_
B. We bump version to 0.2.x and require platform ui 1.8 in this PR, branch out 0.1.x and marking it as non compatible with platform ui 1.8+, and basically will need to instruct 1.8 users to upgrade their content on the fly bundle manually as it is currently set to only install 0.1.x updates

**Comment by sunpietro** [comment]
@andrerom the issue was introduced with the latest release of PlatformUI (it introduced a new version of UDW component). This fix applies to PlatformUI 1.8+.
Previous versions were not affected by the issue.

**Comment by dew326** [comment]
>A. You somehow feature detect if this is 1.7 or 1.8, allowing us to support both with same code base (if not to hard to do, this is my personal preference)

@sunpietro I think it can be done by passing some flag like `forceVisibleMethod` (feel free to use a better name) instead of `visibleMethod` and then in the plugin for the `universalDiscoveryView` on event `activeChange` we should detect if `methods` contains `browse` or `finder`.

**Comment by sunpietro** [comment]
ping @dew326 
@andrerom  The newest fix proposal should keep the compatibility with both versions: 1.7 and 1.8

File: bundle/Resources/public/js/views/plugins/cof-createcontent-universaldiscoveryplugin.js [hunk 1]
```diff
@@ -50,6 +52,30 @@ YUI.add('cof-createcontent-universaldiscoveryplugin', function (Y) {
...
+            if (!event.newVal || !this.get('forceVisibleMethod')) {
+                return;
+            }
+
+            browseMethod = this.get('methods').filter(function (method) {
+                return method.get('id') === 'browse';
+            })[0];
+
+            if (!browseMethod) {
+                this.set('visibleMethod', 'finder');
```
Comment (dew326) [inline bundle/Resources/public/js/views/plugins/cof-createcontent-universaldiscoveryplugin.js]: one method should be always set. If `browse` view exists the `browse` method should be set. 
And also I think we should reset the `forceVisibleMethod`, I don't know if this flag will stay when the UDW is opened again.

**Review by dew326 — CHANGES_REQUESTED** [review]

**Review by andrerom — APPROVED** [review]

**Review by micszo — CHANGES_REQUESTED** [review]
I suppose I have to test this again and on two versions this time. :P I put my +1 on hold and set this to request changes until done.

**Comment by micszo** [comment]
1.8 retest OK.

**Comment by micszo** [comment]
1.7 retest OK.

**Comment by micszo** [comment]
1.8 retest OK.

**Review by micszo — APPROVED** [review]
It works on both versions now. 🎉

**Review by dew326 — APPROVED** [review]

**Comment by andrerom** [comment]
Thanks all :)

@yannickroger / @adib78  I'll tag rc1 and send to QA, they might ask for updated meta tags.

**Pull request merged**